pandas/_libs/lib.pyx

```diff
@@ -2085,11 +2094,19 @@ def maybe_convert_objects(ndarray[object] objects, bint try_float=0,
 
     if not seen.object_:
         if not safe:
-            if seen.null_:
+            if seen.null_ or seen.nan_:
```

Comment:
separate issue is we should change line 2083 to use a DatetimeArray (can be separate PR) or here if it works out.

can you also update the doc-string (well add it really :->) thanks for workign on this.

Reply:
I will leave line 2083 out as it is a separate issue.
Which version of the doc-string shall I modify?